src/router.js

```diff
@@ -35,6 +36,37 @@ const logoutAndRedirect = async (to, from, next) => {
   return next('/login');
 };
 
+const checkAndSubscribeToLobby = async (to, from, next) => {
+  const gameStore = useGameStore();
+  const authStore = useAuthStore();
+  const gameId = parseInt(to.params.gameId);
+  
+  if (Number.isNaN(gameId) || !Number.isFinite(gameId)) {
+    to.meta.error = 'Invalid lobby number';
+    return next();
+  }
+
+  if (!authStore.authenticated) {
+    authStore.redirectGameId = gameId;
+    return next('/login');
+  }
```

Comment:
I think it might be better to use a url param instead of a store variable for the `redirectGameId` because that way we can refresh the page on the `LoginView` and still get back to your game after logging in